lab_2_retrieval_w_bm25/main.py

```diff
@@ -92,6 +163,22 @@ def calculate_tf_idf(tf: dict[str, float], idf: dict[str, float]) -> dict[str, f
 
     In case of corrupt input arguments, None is returned.
     """
+    if not (isinstance(tf, dict) and
+            len(tf) > 0 and
+            all(isinstance(key, str) and isinstance(tf[key], float) for key in tf) and
+            isinstance(idf, dict) and
+            len(idf) > 0 and
+            all(isinstance(key, str) and isinstance(idf[key], float) for key in idf)):
+        return None
+
+    tf_idf = {}
+    for key in tf:
```

Comment:
do we have different keys for tf and idf dictionaries?

Reply:
This is a check: IDF should have all the keys of TF, and if it doesn't, None is returned.

Reply:
Example: the function accidentally takes TF of a text that wasn't used in calculating IDF